projects/plugins/jetpack/modules/widgets/upcoming-events.php

```diff
@@ -120,6 +120,10 @@ public function update( $new_instance, $old_instance ) { // phpcs:ignore Variabl
 	public function widget( $args, $instance ) {
 		require_once JETPACK__PLUGIN_DIR . '/_inc/lib/icalendar-reader.php';
 
+		if ( empty( $instance['feed-url'] ) ) {
```

Comment:
I wonder if this should be added as a new condition below, so one can still have the widget title and all the widget arguments. That would also be an opportunity to display an error message to the site admin explaining that no feed URL is set.

Reply:
Looking at this. Do we have an existing pattern for only displaying something to admins?

Also, in this case, I'm finding the title won't render since $instance is an empty array and [the title depends on the title key being set](https://github.com/Automattic/jetpack/blob/d102d957a4cc1e489bf45391619da8e176ffd10b/projects/plugins/jetpack/modules/widgets/upcoming-events.php#L129-L133).

Reply:
> Do we have an existing pattern for only displaying something to admins?

We don't, I think the way you did it is just fine, personally 👍